socorro/external/sqs/crashqueue.py

```diff
@@ -173,28 +177,51 @@ def get_queue_url(self, queue_name):
         sleep_function=time.sleep,
         module_logger=logger,
     )
-    def build_client(self):
+    def build_client(
+        cls,
+        region,
+        access_key=None,
+        secret_access_key=None,
+        endpoint_url=None,
+        **kwargs,
+    ):
         """Returns a Boto3 SQS Client."""
         # Either they provided ACCESS_KEY and SECRET_ACCESS_KEY in which case
         # we use those, or they didn't in which case boto3 pulls credentials
         # from one of a myriad of other places.
         # http://boto3.readthedocs.io/en/latest/guide/configuration.html#configuring-credentials
         session_kwargs = {}
-        if self.config.access_key and self.config.secret_access_key:
-            session_kwargs["aws_access_key_id"] = self.config.access_key
-            session_kwargs["aws_secret_access_key"] = self.config.secret_access_key
+        if access_key and secret_access_key:
+            session_kwargs["aws_access_key_id"] = access_key
+            session_kwargs["aws_secret_access_key"] = secret_access_key
         session = boto3.session.Session(**session_kwargs)
 
         kwargs = {
             "service_name": "sqs",
-            "region_name": self.config.region,
+            "region_name": region,
         }
-        if self.config.sqs_endpoint_url:
-            kwargs["endpoint_url"] = self.config.sqs_endpoint_url
+        if endpoint_url:
+            kwargs["endpoint_url"] = endpoint_url
 
         return session.client(**kwargs)
 
+    def get_queue_url(self, queue_name):
+        """Returns the SQS queue url for the given queue name
+
+        :arg queue_name: the queue name
+
+        :returns: the url for the queue
+
+        """
+        return self.client.get_queue_url(QueueName=queue_name)["QueueUrl"]
```

Comment:
Unlike the crashstorage classes that use a connectioncontext which then creates a client, this creates the boto client without the intermediary. I almost changed this, but theoretically we're moving to GCP so the SQS code is not long for this world and spending more time on it doesn't make a lot of sense.